kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -724,26 +725,19 @@ class ResidualBasedBlockBuilderAndSolver
 
         unsigned int nthreads = ParallelUtilities::GetNumThreads();
 
-        typedef std::unordered_set < NodeType::DofType::Pointer, DofPointerHasher>  set_type;
-
         KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", ( this->GetEchoLevel() > 2)) << "Number of threads" << nthreads << "\n" << std::endl;
 
         KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", ( this->GetEchoLevel() > 2)) << "Initializing element loop" << std::endl;
 
-        /**
-         * Here we declare three sets.
-         * - The global set: Contains all the DoF of the system
-         * - The slave set: The DoF that are not going to be solved, due to MPC formulation
-         */
-        set_type dof_global_set;
-        dof_global_set.reserve(number_of_elements*20);
+        BaseType::mDofSet = DofsArrayType();
+        BaseType::mDofSet.reserve(number_of_elements*20);
 
         #pragma omp parallel firstprivate(dof_list, second_dof_list)
         {
             const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
 
             // We cleate the temporal set and we reserve some space on them
-            set_type dofs_tmp_set;
+            DofsArrayType dofs_tmp_set;

```

Comment:
this is VERY inefficient. it was effective to insert into this since it was an unordered_set. Being a DofsArrayType you cannot insert into it!!